src/main/java/org/cryptomator/cli/Args.java

```diff
@@ -79,14 +94,33 @@ public int getPort() {
 	}
 
 	public Set<String> getVaultNames() {
-		return vaultPaths.keySet().stream().filter(vaultPasswords::containsKey).map(String.class::cast).collect(Collectors.toSet());
+		Set<String> filteredVaults = vaultPaths.keySet().stream().filter(vaultPasswords::containsKey).map(String.class::cast).collect(Collectors.toSet());
+		filteredVaults.addAll(vaultPaths.keySet().stream().filter(vaultPasswordfiles::containsKey).map(String.class::cast).collect(Collectors.toSet()));
+		return filteredVaults;
 	}
 
 	public String getVaultPath(String vaultName) {
 		return vaultPaths.getProperty(vaultName);
 	}
 
+	public String getVaultPasswordPath(String vaultName) {	return vaultPasswordfiles.getProperty(vaultName); }
+
 	public String getVaultPassword(String vaultName) {
+		if (vaultPasswords.getProperty(vaultName) == null){
+			Path vaultPasswordPath = Paths.get(vaultPasswordfiles.getProperty(vaultName));
+			if (Files.isReadable(vaultPasswordPath) && Files.isRegularFile(vaultPasswordPath)){
+				try (Stream<String> lines = Files.lines(vaultPasswordPath)) {
+					String vaultPassword = lines.findFirst().get().toString();
+					if (vaultPassword != "") {
```

Comment:
Empty string is technically a valid passphrase, even though we don't allow this in the GUI application.
I'd vote to remove this if-statement due to simplicity.